shell/src/main/java/alluxio/cli/ValidateEnv.java

```diff
@@ -0,0 +1,266 @@
+      + "For example, specifying NAME \"master\" or \"ma\" will run both tasks named "
+      + "\"master.rpc.port.available\" and \"master.web.port.available\" but not "
+      + "\"worker.rpc.port.available\".\n"
+      + "If NAME is not given, all tasks for the given TARGET will run.\n";
+
+  private static final Map<ValidationTask, String> TASKS = new HashMap<>();
+
+  private static final String ALLUXIO_MASTER_CLASS = "alluxio.master.AlluxioMaster";
+  private static final String ALLUXIO_WORKER_CLASS = "alluxio.worker.AlluxioWorker";
+  private static final String ALLUXIO_PROXY_CLASS = "alluxio.proxy.AlluxioProxy";
+
+  // port availability validations
+  private static final ValidationTask MASTER_RPC_VALIDATION_TASK = registerTask(
+      "master.rpc.port.available",
+      new PortAvailabilityValidationTask(ServiceType.MASTER_RPC, ALLUXIO_MASTER_CLASS));
+  private static final ValidationTask MASTER_WEB_VALIDATION_TASK = registerTask(
+      "master.web.port.available",
+      new PortAvailabilityValidationTask(ServiceType.MASTER_WEB, ALLUXIO_MASTER_CLASS));
+  private static final ValidationTask WORKER_DATA_VALIDATION_TASK = registerTask(
+      "worker.data.port.available",
+      new PortAvailabilityValidationTask(ServiceType.WORKER_DATA, ALLUXIO_WORKER_CLASS));
+  private static final ValidationTask WORKER_RPC_VALIDATION_TASK = registerTask(
+      "worker.rpc.port.available",
+      new PortAvailabilityValidationTask(ServiceType.WORKER_RPC, ALLUXIO_WORKER_CLASS));
+  private static final ValidationTask WORKER_WEB_VALIDATION_TASK = registerTask(
+      "worker.web.port.available",
+      new PortAvailabilityValidationTask(ServiceType.WORKER_WEB, ALLUXIO_WORKER_CLASS));
+  private static final ValidationTask PROXY_WEB_VALIDATION_TASK = registerTask(
+      "proxy.web.port.available",
+      new PortAvailabilityValidationTask(ServiceType.PROXY_WEB, ALLUXIO_PROXY_CLASS));
+
+  // ssh validations
+  private static final ValidationTask MASTERS_SSH_VALIDATION_TASK = registerTask(
+      "masters.ssh.reachable",
+      new SshValidationTask("masters"));
+  private static final ValidationTask WORKERS_SSH_VALIDATION_TASK = registerTask(
+      "workers.ssh.reachable",
+      new SshValidationTask("workers"));
+
+  private static final Map<String, Collection<ValidationTask>> TARGET_TASKS =
+      initializeTargetTasks();
+
+  private static Map<String, Collection<ValidationTask>> initializeTargetTasks() {
+    Map<String, Collection<ValidationTask>> targetMap = new TreeMap<>();
+    targetMap.put("master", Arrays.asList(
+        MASTER_RPC_VALIDATION_TASK,
+        MASTER_WEB_VALIDATION_TASK,
+        PROXY_WEB_VALIDATION_TASK,
+        MASTERS_SSH_VALIDATION_TASK,
+        WORKERS_SSH_VALIDATION_TASK
+    ));
+    targetMap.put("worker", Arrays.asList(
+        WORKER_DATA_VALIDATION_TASK,
+        WORKER_RPC_VALIDATION_TASK,
+        WORKER_WEB_VALIDATION_TASK,
+        PROXY_WEB_VALIDATION_TASK,
+        MASTERS_SSH_VALIDATION_TASK,
```

Comment:
`MASTERS_SSH_VALIDATION_TASK` is used to verify ssh ports on master node(s) listed in `conf/masters` file are reachable.